packages/@aws-cdk/aws-lambda/lib/alias.ts

```diff
@@ -193,6 +207,31 @@ export class Alias extends QualifiedFunctionBase implements IAlias {
     });
   }
 
+  /**
+   * Enable autoscaling for provisioned concurrency lambda functions. Tracks the predefined metric
+   * LambdaProvisionedConcurrencyUtilization. More information about lambda performance metrics here:
+   * https://docs.aws.amazon.com/lambda/latest/dg/monitoring-metrics.html#monitoring-metrics-concurrency
+   * @param props The properties for autoscaling
+   */
+  public autoScaleProvisionedConcurrency(props: EnableScalingProps): IScalableFunctionAttribute {
+    if (this.scalableAlias) {
+      throw new Error('Autoscaling already enabled for this alias');
+    }
+    // This is a hack to get the ScalableTarget resource to depend on Alias.
+    // this.functionArn depends on the Alias resource. Otherwise, name could easily be this.lambda.functionName
+    const name = this.stack.parseArn(this.functionArn, ':').resourceName!;
+    this.scalableAlias = new ScalableFunctionAttribute(this, 'AliasScaling', {
+      minCapacity: props.minCapacity,
+      maxCapacity: props.maxCapacity,
+      resourceId: `function:${name}:${this.aliasName}`,
+      dimension: 'lambda:function:ProvisionedConcurrency',
+      role: this.scalingRole,
+    });
+
+    // this.scalableAlias.node.tryFindChild('Target')?.node.addDependency(this);
+    return this.scalableAlias;
```

Comment:
The short answer: use `this.functionName` instead of `this.lambda.functionName` for the `resourceId`, from the docs: 

```
functionName:
ARN of this alias

Used to be able to use Alias in place of a regular Lambda.
Lambda accepts ARNs everywhere it accepts function names.
```
This will create the implicit dependency between the `scalableTarget` and the `alias`.

As to why `this.scalableAlias.node.addDependency(this)` created a circular dependency.  The `addDependency` function adds all constructs in the dependency's scope as dependencies to the node, which means all constructs defined in the `alias` scope will be added as dependencies to the `scalableTarget` resource, including the `ScalingPolicy`, which is dependent on `scalableTarget` --> circular dependency.  A picture:
```
ScalableTarget  --dependsOn-> ScalingPolicy --dependsOn-> ScalableTarget
```
Since we only want to add an explicit dependency between the `alias` resource and the `scalableTarget` we need to add only the `CfnAlias ` as dependency:

```ts
// in the constructor
this.alias = new CfnAlias(...);

public autoScaleProvisionedConcurrency(props: EnableScalingProps): IScalableFunctionAttribute {
...
  this.scalableAlias.node.addDependency(this.alias) <- will only add a dependency on the alias cfn resource.
}
```

Since we can use the functionName to create an implicit dependency  there is no need to add an explicit one.

Reply:
This sounds good. Looks similar to the hack that I had but much more elegant :)